src/library/export/engineprimeexportjob.cpp

```diff
@@ -143,7 +143,7 @@ bool tryGetBeatgrid(BeatsPointer pBeats,
 
     int numBeatsToCue = pBeats->numBeatsInRange(firstBeatPlayPos, cueBeatPlayPos);
     const auto firstBarAlignedBeatPlayPos = pBeats->findNBeatsFromPosition(
-            cueBeatPlayPos, numBeatsToCue & ~0x3);
+            cueBeatPlayPos, -1 * (numBeatsToCue & ~0x3));
```

Comment:
What a blunder... this code has been unchanged for a long time, but it mistakenly trying to find the first bar in the track by scanning forwards instead of backwards.

This resulted in the beatgrid being calculated more and more innaccurately the further forward the main cue point was in the track.